RecoVertex/PrimaryVertexProducer/src/DAClusterizerInZ_vect.cc

```diff
@@ -706,6 +709,9 @@ bool DAClusterizerInZ_vect::split(const double beta, track_t& tks, vertex_t& y,
       }
     }
 
+    if ((p1 + p2) < 1.e-100) {
+      continue;
```

Comment:
I think that a similar comment as above applies here as well. This sum is used only in a ratio in `p_{1,2}/(p1+p2)*rho`.
A more appropriate change would perhaps be to directly compute `(p_{1,2}/(p1+p2))*rho`, I'm assuming the compiler would obey the parentheses and compute the ratio first.

Reply:
> A more appropriate change would perhaps be to directly compute `(p_{1,2}/(p1+p2))*rho`, I'm assuming the compiler would obey the parentheses and compute the ratio first.

I see this package is built with
https://github.com/cms-sw/cmssw/blob/5c2d038303b74b35aa5baf58c05bf68c925083f8/RecoVertex/PrimaryVertexProducer/BuildFile.xml#L1
so I wouldn't assume much on how the compiler may decide to reorder the computations.

Reply:
Could we directly check `pk1` and `pk2` below?

```cpp
if (not (std::isfinite(pk1) and std::isfinite(pk2)))
        continue;
```